aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -55,7 +66,7 @@ impl DelayedFieldData {
                     &input,
                     max_value,
                 )?;
-                if result {
+                if result && apply_delta {
```

Comment:
I assume apply_delta = false is the new case, in which case is the behavior supposed to mirror apply_delta = true but without modifying self? Can w add one-liner comment? 

I think apply_delta can and should be statically known parameter: every call is actually statically set true or false, and that way, with generics or macros we should be able to force self to be &mut or & based on the apply_delta value. Then the compiler will prove that when apply_delta = false, there can be no side effects, and it won't be error prone in the future to refactor this function and accidentally forget to guard mutations with apply_delta. It will also avoid the need of tests on apply_delta.

Of course we can also have separate methods, but that'd require duplication significant code.

Reply:
synced up to try and do this in followup, if reasonably readable